xarray/core/common.py

```diff
@@ -474,6 +474,32 @@ def rolling(self, min_periods=None, center=False, **windows):
         Returns
         -------
         rolling : type of input argument
+        
+        Examples
+        --------
+        Create rolling seasonal average of monthly data e.g. DJF, JFM, ..., SON:
+        
+        >>> da = xr.DataArray(np.linspace(0,11,num=12), 
+                              coords=[pd.date_range('15/12/1999',
+                                      periods=12, freq=pd.DateOffset(months=1))], 
+                              dims='time')
+        >>> da
+        <xarray.DataArray (time: 12)>
+        array([  0.,   1.,   2.,   3.,   4.,   5.,   6.,   7., 8.,   9.,  10.,  11.])
+        Coordinates:
+          * time     (time) datetime64[ns] 1999-12-15 2000-01-15 2000-02-15 ...
+        >>> da_avg = da.rolling(time=3).mean()
+        >>> da_avg
+        <xarray.DataArray (time: 12)>
+        array([ nan,  nan,   1.,   2.,   3.,   4.,   5.,   6.,   7.,   8.,   9.,  10.])
+        Coordinates:
+          * time     (time) datetime64[ns] 1999-12-15 2000-01-15 2000-02-15 ...
+        >>> da_avg = da.rolling(time=3).mean().dropna('time')
```

Comment:
Either add a little more exposition here to make it clear why you're showing this or drop it.

Reply:
Are you referring to the output of `da.rolling(time=3).mean()` or the `da.rolling(time=3).mean().dropna('time')` statement?
You originally said it may be worth showing the output before restructuring the result of rolling. I wasn't sure about adding a comment in the middle of an example so i've opted to expand the description of the example. I decided to cut out the `da.rolling(time=3).mean()` line as a believe the output of `da.rolling(time=3).mean().dropna('time')` is cleaner but I understand if you would prefer not to add another function in the example.

Reply:
You can definitely add comments in the middle of examples (e.g., see "Example" for [`np.matmul`](https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.matmul.html)). Just add line breaks before/after and don't preface those lines with `>>>`, e.g.,
```
Add two numbers:

>>> 1 + 1
2

Subtract two numbers:

>>> 1 - 1
0
```

Even though the output is not as clean, my preference is for showing "minimal units" in docstrings so it's entirely clear to users how a function works. That way they'll know to expect NaNs, for example.

So you can also include the example using `dropna()` (with a comment) but we should have the simpler version without it first.